src/vorta/views/repo_add_dialog.py

```diff
@@ -196,43 +132,80 @@ def validate(self):
 
-    def password_listener(self):
+
+class AddRepoWindow(RepoWindow):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.passwordInput = PasswordInput()
+        self.passwordInput.add_form_to_layout(self.repoDataFormLayout)
+
+        self.encryptionComboBox.activated.connect(self.display_backend_warning)
+        self.encryptionComboBox.currentIndexChanged.connect(self.encryption_listener)
+
+        self.display_backend_warning()
+
+    def set_password(self, URL):
+        '''Autofill password from keyring only if current entry is empty'''
+        password = VortaKeyring.get_keyring().get_password('vorta-repo', URL)
+        if password and self.passwordInput.get_password() == "":
+            self.passwordInput.set_error_label(self.tr("Autofilled password from password manager."))
+            self.passwordInput.passwordLineEdit.setText(password)
+            self.passwordInput.confirmLineEdit.setText(password)
+
+    @property
+    def values(self):
+        out = dict(
+            ssh_key=self.sshComboBox.currentData(),
+            repo_url=self.repoURL.text(),
+            password=self.passwordInput.get_password(),
+            extra_borg_arguments=self.extraBorgArgumentsLineEdit.text(),
+            encryption=self.encryptionComboBox.currentData(),
+        )
+        return out
+
+    def encryption_listener(self):
         '''Validates passwords only if its going to be used'''
         if self.values['encryption'] == 'none':
-            self.passwordLabel.setText("")
-            return True
+            self.passwordInput.set_validation_enabled(False)
         else:
-            firstPass = self.passwordLineEdit.text()
-            secondPass = self.confirmLineEdit.text()
-            msg = validate_passwords(firstPass, secondPass)
-            self.passwordLabel.setText(translate('utils', msg))
-            return not bool(msg)
+            self.passwordInput.set_validation_enabled(True)
+
+    def display_backend_warning(self):
+        '''Display password backend message based off current keyring'''
+        if self.encryptionComboBox.currentData() != 'none':
+            self.passwordInput.set_error_label(VortaKeyring.get_keyring().get_backend_warning())
+
+    def run(self):
+        if self.validate() and self.passwordInput.validate():
+            params = BorgInitJob.prepare(self.values)
+            if params['ok']:
+                self.saveButton.setEnabled(False)
+                job = BorgInitJob(params['cmd'], params)
+                job.updated.connect(self._set_status)
+                job.result.connect(self.run_result)
+                QApplication.instance().jobs_manager.add_job(job)
+            else:
+                self._set_status(params['message'])
 
 
-class ExistingRepoWindow(AddRepoWindow):
+class ExistingRepoWindow(RepoWindow):
     def __init__(self):
         super().__init__()
         self.encryptionComboBox.hide()
         self.encryptionLabel.hide()
```

Comment:
These widgets aren't used in this subclass. Therefore it would make more sense not to add them in the superclass.